lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -0,0 +1,99 @@
+"""
+Visibility Transformer implementation.
+"""
+from openedx.core.lib.block_structure.transformer import BlockStructureTransformer, FilteringTransformerMixin
+from xmodule.seq_module import SequenceModule
+from .utils import collect_merged_boolean_field, collect_merged_date_field
+
+
+class HiddenContentTransformer(FilteringTransformerMixin, BlockStructureTransformer):
+    """
+    A transformer that enforces the hide_after_due field on
+    blocks by removing children blocks from the block structure for
```

Comment:
I thought we were making "content is hidden" and "block exists in structure" 2 separate things - is `hide_after_due` supposed to conceal the block's existence by removing it from the structure, or just hide the content?

Maybe I don't fully understand what "removing from the block structure" is doing...